js/src/hooks/useCallbackOnceEffect.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useIsEqualRefValue from './useIsEqualRefValue';
+
+/**
+ * A hook to call a callback function once when a condition is set to true.
+ *
+ * @param {boolean} condition Condition to call the callback.
+ * @param {Function} callback Callback to be executed once when condition is true.
+ * @param  {...any} args Arguments to be passed to the callback.
+ */
+const useCallbackOnceEffect = ( condition, callback, ...args ) => {
+	const calledRef = useRef( false );
+	const argsRefValue = useIsEqualRefValue( args );
```

Comment:
❓ 💅 
Why is this useIsEqualRefValue needed?

Reply:
Because `args` is used as dependency in `useEffect`. Without `useIsEqualRefValue`, `args` will always be a new array in every render, causing the effect to be fired even though the arguments do not change. With `useIsEqualRefValue`, we use the same `args` value when the arguments do not change, so that the effect does not get fired unnecessarily.

Reply:
Interesting, i thought that was checked by the dependencies of useEffect automatically. Or it because the spread operator??

Reply:
Yes, the spread operator is the reason for us here. 

In other cases, if we pass in newly-created objects via function parameter, we should use `useIsEqualRefValue` too before using the object as hook dependencies. For example, see `useApiFetchEffect` below, where we pass in `options` object and wrap it with `useIsEqualRefValue` before using it as `useEffect` hook dependency:

https://github.com/woocommerce/google-listings-and-ads/blob/daa277c2c3de6743086841ad4ab7988872f2f9d9/js/src/hooks/useApiFetchEffect.js#L20-L33